src/fields.cpp

```diff
@@ -213,6 +213,40 @@ fields_chunk::~fields_chunk() {
   if (new_s && new_s->refcount-- <= 1) delete new_s; // delete if not shared
 }
 
+static void split_into_tiles(grid_volume gvol, std::vector<grid_volume> *result) {
+  const size_t base_nowned_min = 12000;
+  if (gvol.nowned_min() < base_nowned_min) {
+    result->push_back(gvol);
+    return;
+  }
+
+  int best_split_point;
+  direction best_split_direction;
+  double left_effort_fraction;
+  gvol.find_best_split(2, false /* fragment_cost */, best_split_point, best_split_direction,
+                       left_effort_fraction);
+  grid_volume left_gvol = gvol.split_at_fraction(false, best_split_point, best_split_direction);
+  split_into_tiles(left_gvol, result);
+  grid_volume right_gvol = gvol.split_at_fraction(true, best_split_point, best_split_direction);
+  split_into_tiles(right_gvol, result);
+  return;
```

Comment:
This is tricky, because `split_at_fraction` is designed to split a domain into *overlapping* chunks, of which the "owned" volumes are non-overlapping.  Also, the whole `find_best_split` machinery is overkill here because we just want to split into half along the longest dimension.

It might be clearer for tiling if we split the whole volume gv into non-overlapping tile (in halves, recurssively), where tile is described by an `<is,ie>` pair of corners, with `is` always an *even* grid point.

Then, when you are doing `step_curl` for a component `c` and a tile `<is,ie>`, you will pass it the something like `max(gv.little_owned_corner0(c), is + iyee_shift(c))` and `ie` for the corners.  This way it will cover the owned volume for `c` intersected with `<is,ie>`